src/app/merge-diff/merge-diff-adapter/merge-diff-adapter.service.ts

```diff
@@ -27,8 +31,39 @@ import { MdmResourcesService } from '@mdm/modules/resources';
   providedIn: 'root'
 })
 export class MergeDiffAdapterService {
+  constructor(private resources: MdmResourcesService, private http: HttpClient ) {}
+
+  loadCatalogueItemDetails(id: Uuid, domainType: ModelDomainType) : Observable<any> {
+    switch (domainType) {
+      case ModelDomainType.DataModels:
+        return this.resources.dataModel.get(id);
+      case ModelDomainType.ReferenceDataModels:
+        return this.resources.referenceDataModel.get(id);
+      default:
+        break;
+    }
+  }
+
+  retrieveMainBranch(domainType: ModelDomainType, sourceId: Uuid)
+  {
+    return this.resources.versioning.currentMainBranch(domainType, sourceId);
+  }
+
+  // getMergeDiff(sourceId: Uuid, targetId: Uuid): Observable<MergeItem[]> {
+  //   return this.resources.
+  //      versioning.
+  //      mergeDiff(sourceId, targetId,targetId)
+  //     .pipe(
+  //       catchError(() => {return EMPTY;}),  // Common error handling (maybe)
+  //       map((response: VersioningResponse) => response.body)  // Map to types we care about
+  //     );
+  // }
+
+  getMergeDiff() : Observable<any>
+  {
+    return this.http.get('../../../assets/newStyle.json');
```

Comment:
If this is a local file there is another way to load in JSON from file, without needing this `HttpClient` injected.

Basically you can import a JSON file just like a JS/TS file/module:

```ts
import * from './test-file.json`
```

I think our Angular/TypeScript config files allow this, if not there may be some switches involved.

The JSON is then imported just like an exported object literal.

Reply:
this will be removed once the BE is ready.